orderbook/src/account_balances.rs

```diff
@@ -0,0 +1,195 @@
+use anyhow::{Context, Result};
+use ethcontract::{batch::CallBatch, errors::MethodError, Http, Web3};
+use futures::future::{join, join_all, ok, FutureExt as _};
+use std::collections::HashMap;
+use std::sync::Mutex;
+
+use contracts::IERC20;
+use primitive_types::{H160, U256};
+
+#[async_trait::async_trait]
+pub trait BalanceFetching: Send + Sync {
```

Comment:
Shouldn't the update function which is pub on the struct be part of the trait too?